packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -554,212 +594,234 @@ function basicStateReducer<S>(state: S, action: BasicStateAction<S>): S {
   return typeof action === 'function' ? action(state) : action;
 }
 
-export function useContext<T>(
+function mountContext<T>(
   context: ReactContext<T>,
   observedBits: void | number | boolean,
 ): T {
   if (__DEV__) {
     currentHookNameInDev = 'useContext';
-    createWorkInProgressHook();
+    mountWorkInProgressHook();
     currentHookNameInDev = null;
   }
-  // Ensure we're in a function component (class components support only the
-  // .unstable_read() form)
-  resolveCurrentlyRenderingFiber();
   return readContext(context, observedBits);
 }
 
-export function useState<S>(
-  initialState: (() => S) | S,
-): [S, Dispatch<BasicStateAction<S>>] {
+function updateContext<T>(
+  context: ReactContext<T>,
+  observedBits: void | number | boolean,
+): T {
   if (__DEV__) {
-    currentHookNameInDev = 'useState';
+    currentHookNameInDev = 'useContext';
+    updateWorkInProgressHook();
+    currentHookNameInDev = null;
   }
-  return useReducer(
-    basicStateReducer,
-    // useReducer has a special case to support lazy useState initializers
-    (initialState: any),
-  );
+  return readContext(context, observedBits);
 }
 
-export function useReducer<S, A>(
+function mountReducerImpl<S, A>(
```

Comment:
The weird inner branch didn't concern me because I figured it would go away once we change the lazy init API to the new one, but you're right, this is small enough that we should just inline it.

Reply:
Ugh. GitHub showed me an old commit. Looks fixed it.